### PR TITLE
Maven publish with GitHub Actions and Gradle 7 compatibility

### PR DESCRIPTION
**Pull request by kaisalmen**
The gradle build instruction use a an old maven plugin which is incompatible with Gradle 7. This PR replaces the it with `maven-publish` plugin and introduces the `sign` pluging.

Further gradle 7 build deprecation issues have been fixed (compile step and duplication strategy).

GitHub Actions deploy build artifacts on every successful master builds (enhancement to main.yml)
GitHub Actions deploy release artifacts on release via GitHub and when the version does not contain "SNAPSHOT".

### CODE REVIEW

**Comment by kaisalmen** [comment]
Ok, a pull-request should trigger a build, but no deployment this needs to be fixed.

**Comment by spoenemann** [comment]
@kaisalmen you can add a condition like this:
https://github.com/eclipse/openvsx/blob/3c17e06825484655a5d5904df73191c07f81530e/.github/workflows/main.yml#L45

**Comment by kaisalmen** [comment]
@spoenemann thanks and I need to split the steps as well, build always, condition for publish

**Comment by kaisalmen** [comment]
Is there a specific reason you have the repository name in the if condition? Isn't the branch name sufficient or is there another reason I am not aware of?

**Comment by spoenemann** [comment]
> Is there a specific reason you have the repository name in the if condition?

To skip the publishing step when someone forks the repository.

File: yang-lsp/gradle/publishing.gradle [hunk 1]
```diff
@@ -0,0 +1,109 @@
+java {
+	withJavadocJar()
+	withSourcesJar()
```
Comment (spoenemann) [inline yang-lsp/gradle/publishing.gradle:3]: Are these automatically generated? We currently have hand-written tasks for this in java-compiler-settings.gradle, maybe we can remove those?

**Review by spoenemann — APPROVED** [review]
Thank you!

File: yang-lsp/gradle/publishing.gradle (hunk 1 above)
Comment (kaisalmen) [reply yang-lsp/gradle/publishing.gradle:3]: They are not needed here, I will remove it.

**Pull request merged**